03-transactions.md

```diff
@@ -827,6 +836,101 @@ at each bucket is a prefix of the desired index.
 
 # Appendix A: Expected Weights
 
+## Expected Weight of the Funding Transaction (v2 Channel Establishment)
+
+The *expected weight* of a funding transaction is calculated as follows:
+
+      inputs: 41 bytes
+        - previous_out_point: 36 bytes
+          - hash: 32 bytes
+          - index: 4 bytes
+        - var_int: 1 byte
+        - script_sig: 0 bytes
+        - witness <---- Cost for "witness" data calculated separately.
+        - sequence: 4 bytes
+
+      non_funding_outputs: 9 bytes + `scriptlen`
+        - value: 8 bytes
+        - var_int: 1 byte <---- assuming a standard output script
+        - script: `scriptlen`
+
+      funding_output: 43 bytes
+        - value: 8 bytes
+        - var_int: 1 byte
+        - script: 34 bytes
+          - OP_0: 1 byte
+          - PUSHDATA(32-byte-hash): 33 bytes
+
+Multiplying non-witness data by 4 results in a weight of:
+
+      // transaction_fields = 10 (version, input count, output count, locktime)
+      // segwit_fields = 2 (marker + flag)
+      // funding_transaction = 43 + num_inputs * 41 + num_outputs * 9 + sum(scriptlen)
+      funding_transaction_weight = 4 * (funding_transaction + transaction_fields) + segwit_fields
+
+      witness_weight = sum(witness_len)
+
+      overall_weight = funding_transaction_weight + witness_weight
+
+### Calculating Fees for Collaborative Transaction Construction
+
+Every participant in a collaborative transaction covers the fees for
+their own inputs and outputs. The initiator also provides funds to
+cover for the common transaction fields.
+
+Upon successful exchange of `tx_complete` messages, each peer must at
+least cover a minimum estimated fee. Here is how to calculate the minimum
+fee for each participant.
+
+The minimum witness weight for an input is 107.
```

Comment:
I believe this is wrong, the witness weight when spending a P2TR input can be smaller than that, right?

Reply:
not sure it's that helpful to even have a specific number here, since it's really a variable. Could give some prototypical examples and leave a variable in the following sections?

Reply:
> I believe this is wrong, the witness weight when spending a P2TR input can be smaller than that, right?

yes, p2tr key path with sighash default has a witness weight of 66 wu, but even smaller weights are possible. an anyone can spend p2wsh (`OP_1`) can be spent with 4 wu of witness

Reply:
This is an issue because the `tx_complete` section in Bolt 2 contains a requirement to fail the negotiation if the feerate doesn't seem to be met based on the minimum witness weight provided here, which means we're currently rejecting funding attempts that use p2tr inputs. I think that we should relax this requirement and update these Bolt 3 sections as suggested by @instagibbs

Reply:
agreed, it's out of date at this point. the goal was to prevent a peer from underpaying on the fees for their witnesses. there's no way to know a priori what witness weight will be tho, so we enforced a minimum payment for it in the protocol.

Which seems is a bit unnecessary: if they're going to underpay, they're going to underpay.